code/scripting/api/libs/ui.cpp

```diff
@@ -798,12 +798,18 @@ ADE_INDEXER(l_Briefing_Goals,
 	"number Index",
 	"Array of goals",
 	"mission_goal",
-	"goal handle, or invalid handle if index is invalid")
+	"goal handle, or nil if index is invalid")
 {
 	int idx;
 	if (!ade_get_args(L, "*i", &idx))
 		return ade_set_error(L, "s", "");
 
+	// convert from lua index
+	idx--;
+
+	if ((idx < 0) || idx >= (int)Mission_goals.size())
+		return ADE_RETURN_NIL;
```

Comment:
Other indexers in the codebase return an invalid handle when the index is out of range, so for consistency's sake this one should do the same thing.  In the Personas indexer, for example, the code is:
```
if (idx < 0 || idx >= Num_personas)
		return ade_set_args(L, "o", l_Persona.Set(-1));
```
The -1 index represents an invalid persona.  Then there is a separate isValid function:
```
ADE_FUNC(isValid, l_Persona, NULL, "Detect if the handle is valid", "boolean", "true if valid, false otherwise")
{
	int idx = -1;

	if (!ade_get_args(L, "o", l_Persona.Get(&idx)))
		return ADE_RETURN_FALSE;

	return ade_set_args(L, "b", idx >= 0 && idx < Num_personas);
}
```

Reply:
You sure are a needy little one, aren't you. Done.

Reply:
The `l_Goals` type still needs an `isValid` function.

Reply:
that's a whole lot of a code for something SCPUI _does not_ need and does not use.

Reply:
You and many others have cursed the inconsistency in other parts of the codebase, please let's not introduce more. An `isValid` function on lua objects of all types is completely reasonable and appropriate.

Please also bear in mind that features should not *only* relate to whatever use you have for them, and that other people may want to use this in other ways more than just SCPUI. Modders can use the FSO in all sorts of creative ways, and things should be kept as general purpose as possible, especially for something as basic as an `isValid` function.

Reply:
I fully expect SCP to go on a search to make sure every single indexer has this function.

Reply:
Sometimes that's what happens.  The two goals here are a) consistency with the rest of the Lua API, and b) handling any possible scenario.  One of those scenarios is checking for a valid handle.